patroni/postgresql.py

```diff
@@ -1254,10 +1274,15 @@ def _get_connection_cursor(**kwargs):
 
     @contextmanager
     def _get_replication_connection_cursor(self, host='localhost', port=5432, database=None, **kwargs):
-        with self._get_connection_cursor(host=host, port=int(port), database=database or self._database, replication=1,
-                                         user=self._replication['username'], password=self._replication['password'],
-                                         connect_timeout=3, options='-c statement_timeout=2000') as cur:
-            yield cur
+        if self._replication.get('username', None) and self._replication.get('password', None):
+            with self._get_connection_cursor(host=host, port=int(port), database=database or self._database, replication=1,
```

Comment:
Here we can leave mostly the original code. The only adjustment needed is:
```python
password=self._replication.get('password')
```

If the password is not present `psycopg2.connect` will receive `password=None` and basically ignore it:
```python
c = psycopg2.connect(host='localhost', user='postgres', dbname='postgres', password=None)
```

The above line works without any problems.